bliss/datasets/simulated.py

```diff
@@ -20,7 +20,9 @@ def __init__(self, cfg: DictConfig):
 
         self.n_batches = cfg.dataset.params.n_batches
         self.batch_size = cfg.dataset.params.batch_size
-        self.image_decoder = ImageDecoder(**cfg.model.decoder.params)
+        self.image_decoder = ImageDecoder(**cfg.model.decoder.params).to(
+            cfg.dataset.params.generate_device
+        )
```

Comment:
Since this `image_decoder` is not part of the training model. Pytorch-lightning won't move it to gpu even if the model is trained on gpu. It would be initialized on cpu and stays on cpu unless we don move it manually. Both are tested and speed change is not significantly different but generating on cpu does create a bottleneck.